map_test.go

```diff
@@ -142,6 +138,22 @@ func TestMatchEmptyOrDeleted(t *testing.T) {
 	}
 }
 
+func TestConvertDeletedToEmptyAndFullToDeleted(t *testing.T) {
+	testCases := []struct {
+		ctrls    uint64
+		expected uint64
+	}{
+		{0xff11fe2280334455, 0x80fe80fe80fefefe},
```

Comment:
[nit] This is very obscure, we could instead just randomly set up each slot as either empty/full/deleted and then verify that `matchXX` returns what we expect on the post-conversion result.

Reply:
Done.